src/routes/login/+page.svelte

```diff
@@ -65,21 +65,28 @@
         signInSuccessWithAuthResult(authResult, redirectUrl) {
           awaitingRedirect = true;
           setWillAttempLogin(true);
-          fetch('/login/sessionCookie', { method: 'POST', body: JSON.stringify({ idToken: authResult.user.accessToken }) }).then(() => {
-            if (!actionParam && authResult.additionalUserInfo.isNewUser) {
-              gtag('event', 'new_account');
-              goto(`/account${$page.url.search || '?'}&action=choosePlan`);
-            } else if (isAppRedirect) {
-              fetch('/login/customToken', { method: 'POST', body: JSON.stringify({ idToken: authResult.user.accessToken }) }).then(
-                async (response) => {
-                  const token = (await response.json()).customToken;
-                  window.location.replace(redirectBuilder(authResult.user, token));
-                },
-              );
-            } else {
-              goto(redirectBuilder(authResult.user), { replaceState: true });
-            }
-          });
+          if (authResult.additionalUserInfo.isNewUser) gtag('event', 'new_account');
+          (async () => {
+            // Sets a session cookie on the browser
+            const sessionCookiePromise = fetch('/login/sessionCookie', {
+              method: 'POST',
+              body: JSON.stringify({ idToken: authResult.user.accessToken }),
+            });
+            // Now that we have logged in, the customToken store in firebase.ts will be updated, so we wait for that before redirecting
+            const unsubscribe = customLoginToken.subscribe(async (customToken) => {
+              if (customToken) {
+                setTimeout(() => unsubscribe()); // Make sure this doesn't fire again after the first time the token is updated. SetTimeout is to ensure unsubscribe is defined, because this callback may get called immediately
+                await sessionCookiePromise; // Make sure this finished to set the login cookie before redirecting
```

Comment:
This was previously just calling `unsubscribe()`. However, there is a case where this callback gets called immediately, if the value of `customToken` is already defined by the time we get to line 76. In that case, `unsubscribe` is not defined yet since line 76 needs to finish running for that to work. By adding the `setTimeout`, we guarantee that it runs on the next tick. Honestly considered removing the unsubscribe altogether, but I am concerned in that case that the handler would stay attached to the store and may run again if the value were to update, causing a really weird future bug